src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -6355,6 +6408,416 @@ register_handler(int sig, struct sigaction *old_handler)
 	}
 }
 
+/* free all EQs allocated for aio */
+
+static void
+free_aio_ctx(void)
+{
+	int i;
+
+	for (i = 0; i < aio_eq_count; i++) {
+		daos_eq_destroy(aio_eq_list[i].eq, 0);
+		D_MUTEX_DESTROY(&aio_eq_list[i].lock);
+	}
+	aio_eq_count = 0;
+}
+
+/* aio functions return negative errno in case of failure */
+static int
+create_ev_eq_for_aio(aio_ctx_t *aio_ctx);
+int
+io_setup(int maxevents, io_context_t *ctxp)
+{
+	int        rc;
+	aio_ctx_t *aio_ctx_obj;
+
+	if (next_io_setup == NULL) {
+		next_io_setup = dlsym(RTLD_NEXT, "io_setup");
+		D_ASSERT(next_io_setup != NULL);
+	}
+
+	rc = next_io_setup(maxevents, ctxp);
+	if (rc < 0)
+		return rc;
+
+	D_ALLOC(aio_ctx_obj, sizeof(aio_ctx_t));
```

Comment:
This is a bug I think, it should read as below so that the allocation is the size of the struct rather than the pointer.  To avoid this call of bug we have the `D_ALLOC_PTR` macro which does the work for you, please use it in this case.
```suggestion
	D_ALLOC(aio_ctx_obj, sizeof(*aio_ctx_t));
```
```suggestion
	D_ALLOC_PTR(aio_ctx_obj);
```

Reply:
Thank you! I changed it to "D_ALLOC_PTR(aio_ctx_obj);" for better readability although "D_ALLOC(aio_ctx_obj, sizeof(aio_ctx_t));" is correct. aio_ctx_t is struct not struct pointer.